services/QuillLMS/app/workers/teacher_notifications/send_notification_worker.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+module TeacherNotifications
+  class SendNotificationWorker
+    include Sidekiq::Worker
+
+    def perform(activity_session_id)
+      @activity_session = find_activity_session(activity_session_id)
+      return unless @activity_session
+
+      TeacherNotification.transaction do
```

Comment:
Does this need to be wrapped in a transaction? We have a lot of DB contention on locks, so wondering if this could be removed. Maybe consider breaking into 3 jobs if the concern is keeping the job idempotent.

Reply:
This is a good point.  Combined with your note below about expansion, I think it's probably a good idea to split it into multiple workers.  That said, each of the workers will be in a position where they write a variable number of records (based on how many teachers and co-teachers there are for the class an even takes place in), and I'm a little worried about one of those records saving, the job crashing, then re-running and doubling notifications for one of the teachers but not the other.  Or getting stuck in a retry loop where it keeps adding a record to one teacher, but the other always fails.  Maybe it's better to just take that risk than to add a transaction block, but that's what was on my mind when I added the transaction.

Thougths?